src/nspawn/nspawn-network.c

```diff
@@ -561,13 +568,13 @@ int setup_macvlan(const char *machine_name, pid_t pid, char **ifaces) {
                 if (r < 0)
                         return log_error_errno(r, "Failed to add netlink interface index: %m");
 
-                n = strjoin("mv-", *i);
+                n = strdup(*b);
```

Comment:
if these aren't modified anymore (i.e. no "mv-" prefix added here) then there's no need to allocate any memory. hence remove n and a from this loop, and use i/b directly here.

Reply:
We still need to keep a copy for initial interface name in case it is shortened.
(see last modification)